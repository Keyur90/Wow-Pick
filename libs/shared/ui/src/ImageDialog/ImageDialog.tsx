import React, { useState, useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Dialog, DialogTitle, DialogContent, DialogActions, Typography, Button } from '@material-ui/core';

interface ImageDialogPropTypes {
  isOpen: boolean;
  onToggle: (force?: boolean) => void;
  imageUrl: string;
  errorText?: string;
}

const useStyles = makeStyles((theme) => ({
  imageContainer: {
    width: 250,
    height: 250,
    '& img': {
      width: '100%',
      maxWidth: 250,
      height: 'auto',
    },
  },
  bodyText: {
    color: theme.palette.common.black,
  },
}));

const useDialogContentStyles = makeStyles((theme) => ({
  root: {
    padding: theme.spacing(0, 3),
  },
}));

const useDialogActionStyles = makeStyles((theme) => ({
  root: {
    padding: theme.spacing(3),
  },
}));

const ImageDialog: React.FC<ImageDialogPropTypes> = ({ isOpen, onToggle, imageUrl, errorText }) => {
  const classes = useStyles();
  const dialogContentClasses = useDialogContentStyles();
  const dialogActionClasses = useDialogActionStyles();
  const [hasError, setHasError] = useState(false);

  const handleError = () => {
    if (errorText) {
      setHasError(true);
    }
  };

  useEffect(() => {
    if (isOpen) {
      setHasError(false);
    }
  }, [isOpen]);

  return (
    <Dialog onClose={() => onToggle(false)} open={isOpen}>
      {hasError && <DialogTitle>Image not available</DialogTitle>}
      <DialogContent classes={dialogContentClasses}>
        {!hasError ? (
          <div className={classes.imageContainer}>
            <img src={imageUrl} alt="" onError={handleError} />
          </div>
        ) : (
          <Typography variant="body1" className={classes.bodyText}>
            {errorText}
          </Typography>
        )}
      </DialogContent>
      <DialogActions classes={dialogActionClasses}>
        <Button color="secondary" variant="outlined" onClick={() => onToggle(false)} size={'large'}>
          Close
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export { ImageDialog };
