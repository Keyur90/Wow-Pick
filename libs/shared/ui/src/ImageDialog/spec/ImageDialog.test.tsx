import React from 'react';
import { render } from '@testing-library/react';
import { ImageDialog } from '../ImageDialog';

test('renders ImageDialog', async () => {
  const { container } = render(
    <ImageDialog
      isOpen={true}
      onToggle={() => null}
      imageUrl={'https://cdn0.woolworths.media/content/wowproductimages/medium/266869.jpg'}
    />,
    {
      container: document.body,
    }
  );

  expect(container).toMatchSnapshot();
});
