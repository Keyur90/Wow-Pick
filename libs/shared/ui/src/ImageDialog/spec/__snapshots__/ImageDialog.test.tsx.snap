// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ImageDialog 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    class="MuiDialog-root"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div
          class="MuiDialogContent-root makeStyles-root-3"
        >
          <div
            class="makeStyles-imageContainer-1"
          >
            <img
              alt=""
              src="https://cdn0.woolworths.media/content/wowproductimages/medium/266869.jpg"
            />
          </div>
        </div>
        <div
          class="MuiDialogActions-root makeStyles-root-4 MuiDialogActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Close
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
